src/NServiceBus.Metrics.ServiceControl/ServiceControlReporting/QueueLength/SendQueueLengthReportToServiceControl.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using NServiceBus;
+using NServiceBus.Logging;
+using NServiceBus.Metrics.ServiceControl;
+using NServiceBus.Metrics.ServiceControl.ServiceControlReporting;
+using NServiceBus.Transports;
+using NServiceBus.Unicast;
+
+class SendQueueLengthReportToServiceControl
+{
+    QueueLengthReport report;
+    ISendMessages dispatcher;
+    Dictionary<string, string> headers;
+    string destination;
+    TimeSpan ttbr;
+
+    public SendQueueLengthReportToServiceControl(ISendMessages dispatcher, ReportingOptions options, Dictionary<string, string> headers, QueueLengthReport report)
+    {
+        this.dispatcher = dispatcher;
+        this.headers = headers;
+        destination = options.ServiceControlMetricsAddress;
+        ttbr = options.TimeToBeReceived;
+        this.report = report;
+    }
+
+    public void SendNow()
+    {
+        report.TimeStamp = DateTime.UtcNow.Ticks;
+        var stringBody = SimpleJson.SerializeObject(report);
```

Comment:
If we introduce a `ConcurrentDictionary` for report, this might lock writers when serializing (if SimpleJson accesses `Keys`, or `Values`). Nothing serious, just noticing.

Reply:
I think that should be OK. I wanted to avoid creating a new dictionary every time. I checked and SimpleJson can serialize ConcurrentDictionary but it doesn't do it very efficiently. I was hoping to get 

```js
{
  "queueName": length,
  "queueName": length
}
```

Instead I get

```js
{
  {
    "key": "queueName", 
    "value": length
  },
  { 
    "key": "queueName",
    "value": length
  }
}
```

The same is true for the non-concurrent approach.

I wonder if it is worth hand-serializing the dictionary given that the other end is using Json.NET to deserialize the messages. 

Alternatively do we fall back on the single input queue approach and tackle this problem later.

Reply:
SimpleJSON is... simple ;) We could hand-serialize it. The only reason to have SimpleJSON was not taking a dependency on JSON.NET.